cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaConnectApiImpl.java

```diff
@@ -63,34 +64,41 @@ public KafkaConnectApiImpl(Vertx vertx) {
+            httpClient.request(HttpMethod.PUT, port, host, path, request -> {
+                if (request.succeeded()) {
+                    request.result().setFollowRedirects(true)
+                            .putHeader("Accept", "application/json")
+                            .putHeader("Content-Type", "application/json")
+                            .putHeader("Content-Length", String.valueOf(data.length()))
+                            .write(data);
+                    request.result().send(response -> {
+                        if (response.succeeded()) {
+                            if (response.result().statusCode() == 200 || response.result().statusCode() == 201) {
+                                response.result().bodyHandler(buffer -> {
+                                    try {
+                                        Map t = mapper.readValue(buffer.getBytes(), Map.class);
+                                        log.debug("Got {} response to PUT request to {}: {}", response.result().statusCode(), path, t);
+                                        result.complete(t);
+                                    } catch (IOException e) {
+                                        result.fail(new ConnectRestException(response.result(), "Could not deserialize response: " + e));
+                                    }
+                                });
+                            } else {
+                                // TODO Handle 409 (Conflict) indicating a rebalance in progress
+                                log.debug("Got {} response to PUT request to {}", response.result().statusCode(), path);
+                                response.result().bodyHandler(buffer -> {
+                                    JsonObject x = buffer.toJsonObject();
+                                    result.fail(new ConnectRestException(response.result(), x.getString("message")));
+                                });
+                            }
+                        } else {
+                            result.tryFail(response.cause());
                         }
                     });
                 } else {
-                    // TODO Handle 409 (Conflict) indicating a rebalance in progress
-                    log.debug("Got {} response to PUT request to {}", response.statusCode(), path);
-                    response.bodyHandler(buffer -> {
-                        JsonObject x = buffer.toJsonObject();
-                        result.fail(new ConnectRestException(response, x.getString("message")));
-                    });
+                    result.fail(request.cause());
                 }
-            })
-            .exceptionHandler(result::tryFail)
-            .setFollowRedirects(true)
-            .putHeader("Accept", "application/json")
-            .putHeader("Content-Type", "application/json")
-            .putHeader("Content-Length", String.valueOf(data.length()))
-            .write(data)
-            .end());
```

Comment:
In some places you have called `request.result().end();` and others like here you don't seem to `.end()` the request? I think you need (to be safe) to explicitly end all the requests. Unless vertx 4 does things differently?